lib/giphy/giphy.go

```diff
@@ -0,0 +1,75 @@
+package giphy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type GiphyPicker struct {
+	ApiKey string
+}
+
+type GiphyDownSizedLarge struct {
+	URL string `json:"url"`
+}
+
+type GiphyImageEntry struct {
+	Image GiphyDownSizedLarge `json:"downsized_large"`
+}
+
+type GiphyDataItem struct {
+	Images GiphyImageEntry `json:"images"`
+}
+
+type GiphyAPIResponse struct {
+	Data []GiphyDataItem `json:"data"`
+}
+
+func NewPicker() (*GiphyPicker, error) {
+	giphy_api_key := os.Getenv("GIPHY_API_KEY")
+	if giphy_api_key == "" {
+		return nil, fmt.Errorf("Missing Giphy API Key")
+	}
+
+	return &GiphyPicker{
+		ApiKey: giphy_api_key,
+	}, nil
+}
+
+func (p *GiphyPicker) GetRandomGiphy(s string) (string, error) {
+	roffset := rand.Intn(5)
```

Comment:
I used this as a way to generate a random pic everytime for the movie, it's OK, but the larger the number the more irrelevant the movie gif

Reply:
yeah, this is a WIP, just wanted to show @willabides my progress